core/src/main/java/org/bitcoinj/crypto/DeterministicKey.java

```diff
@@ -186,18 +186,26 @@ private DeterministicKey(@Nullable BigInteger priv, LazyECPoint pub, int depth,
         this.parent = parent;
         this.parentFingerprint = ascertainParentFingerprint(parent, parentFingerprint);
         this.chainCode = Arrays.copyOf(chainCode, chainCode.length);
-        this.childNumberPath = Objects.requireNonNull(hdPath);
+        this.childNumberPath = Objects.requireNonNull(hdPath).asPartial();
```

Comment:
I'm not sure if in these cases we should keep the `requireNonNull()` – a NPE will be thrown anyway. But it certainly doesn't hurt either.